site/src/pages/WorkspacesPage/WorkspacesPage.tsx

```diff
@@ -22,20 +23,61 @@ const WorkspacesPage: FC = () => {
     ...pagination,
     ...filter,
   })
+
   const updateWorkspace = useWorkspaceUpdate(queryKey)
 
+  const [displayedWorkspaces, setDisplayedWorkspaces] = useState<Workspace[]>(
+    [],
+  )
+
+  useEffect(() => {
+    const fetchedWorkspaces = data?.workspaces || []
+    if (fetchedWorkspaces) {
+      if (displayedWorkspaces.length === 0) {
+        setDisplayedWorkspaces(fetchedWorkspaces)
+      } else {
+        // Merge the fetched workspaces with the displayed onws, without changing the order of the existing items
+        const mergedItems = displayedWorkspaces
+          .map((item) => {
+            const fetchedItem = fetchedWorkspaces.find(
+              (fItem) => fItem.id === item.id,
+            )
+
+            if (!fetchedItem) {
+              return null
+            }
+
+            // If the fetched item already exists, update its data without changing its position
+            return { ...item, ...fetchedItem }
+          })
+          .filter((item) => item !== null) // Remove the removed items (null values)
+
+        // Add new items to the beginning of the list
+        fetchedWorkspaces.forEach((fetchedItem) => {
+          if (!mergedItems.some((item) => item?.id === fetchedItem.id)) {
+            mergedItems.unshift(fetchedItem)
+          }
+        })
```

Comment:
Ok, I understand the "hack", but I'm curious how it behaves while changing pages or playing with the filter bar.